src/hooks/captureInterval/useCaptureInterval.ts

```diff
@@ -0,0 +1,183 @@
+import { modifyDraftSpec } from 'api/draftSpecs';
+import {
+    useEditorStore_persistedDraftId,
+    useEditorStore_queryResponse_draftSpecs,
+    useEditorStore_queryResponse_mutate,
+} from 'components/editor/Store/hooks';
+import { debounce, omit } from 'lodash';
+import {
+    Dispatch,
+    SetStateAction,
+    useCallback,
+    useEffect,
+    useMemo,
+    useRef,
+    useState,
+} from 'react';
+import { logRocketConsole } from 'services/shared';
+import { CustomEvents } from 'services/types';
+import { useBindingStore } from 'stores/Binding/Store';
+import { useFormStateStore_setFormState } from 'stores/FormState/hooks';
+import { FormStatus } from 'stores/FormState/types';
+import { Schema } from 'types';
+import { hasLength } from 'utils/misc-utils';
+import { formatCaptureInterval } from 'utils/time-utils';
+import { DEFAULT_DEBOUNCE_WAIT } from 'utils/workflow-utils';
+import {
+    CAPTURE_INTERVAL_RE,
+    NUMERIC_RE,
+    POSTGRES_INTERVAL_RE,
+} from 'validation';
+
+export default function useCaptureInterval() {
+    // Binding Store
+    const captureInterval = useBindingStore((state) => state.captureInterval);
+    const setCaptureInterval = useBindingStore(
+        (state) => state.setCaptureInterval
+    );
+
+    // Draft Editor Store
+    const draftId = useEditorStore_persistedDraftId();
+    const draftSpecs = useEditorStore_queryResponse_draftSpecs();
+    const mutateDraftSpecs = useEditorStore_queryResponse_mutate();
+
+    // Form State Store
+    const setFormState = useFormStateStore_setFormState();
+
+    const [serverUpdateRequired, setServerUpdateRequired] = useState(false);
+
+    const draftSpec = useMemo(
+        () =>
+            draftSpecs.length > 0 && draftSpecs[0].spec ? draftSpecs[0] : null,
+        [draftSpecs]
+    );
+
+    const applyCaptureInterval = useCallback(
+        async (interval: string) => {
+            if (!mutateDraftSpecs || !draftId || !draftSpec) {
+                logRocketConsole(CustomEvents.CAPTURE_INTERVAL_ERROR, {
+                    draftIdMissing: !draftId,
+                    draftSpecMissing: !draftSpec,
+                    mutateMissing: !mutateDraftSpecs,
+                });
+
+                return Promise.resolve();
+            }
+
+            let spec: Schema = draftSpec.spec;
+            const originalInterval = draftSpec.spec?.interval;
+
+            if (!hasLength(interval)) {
+                spec = omit(spec, 'interval');
+            } else {
+                const strippedInterval = interval.endsWith('i')
+                    ? interval.slice(0, interval.length - 1)
+                    : interval;
```

Comment:
If we are doing stuff like this then the `i` needs to be a const and this kind of action should go into the utils.